src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -359,4 +388,18 @@ module.exports = class LevelPlane extends Array {
     }
   }
 
+  /**
+  * Checking power state for objects that are powered by redstone.
+  */
+  powerCheck(position) {
+    let amIPowered = false;
+    this.getOrthogonalPositions(position).forEach(orthogonalPosition => {
```

Comment:
Nit: you can use [`Array.prototype.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here, which will also bail out early if the callback returns true.

Reply:
I was looking at the stackoverflow bits about bailing out of loops in JS (basically just having a return statement within it to get out of the function entirely) but it wasn't working. Chances are that's just my being used to C++ not JS, and there being a built in way to achieve that doesn't surprise me. 👍 

If we want to make that function more streamlined I'm fine with it, but it does seem to have the desired functionality in its current state.

Reply:
I prefer this for readability:

```js
powerCheck(position) {
  return this.getOrthogonalPositions(position).some(orthogonalPosition => (
    this[this.coordinatesToIndex(orthogonalPosition)].isRedstone &&
    this[this.coordinatesToIndex(orthogonalPosition)].isPowered)
  );
}
```